common/src/main/java/com/gregtechceu/gtceu/integration/GTOreVeinWidget.java

```diff
@@ -0,0 +1,138 @@
+        this.dimensionFilter = fluid.getDimensionFilter();
+        this.dimensions = dimensions();
+        this.range = "NULL";
+        setClientSideWidget();
+        setupBaseGui(fluid);
+        setupText();
+    }
+
+    private String range(GTOreDefinition oreDefinition) {
+        HeightProvider height = oreDefinition.getRange().height;
+        int maxHeight = height instanceof UniformHeight ? ((UniformHeight) height).maxInclusive.resolveY(null) : 0;
+        int minHeight = height instanceof UniformHeight ? ((UniformHeight) height).minInclusive.resolveY(null) : 0;
+        return String.format("%d - %d", minHeight, maxHeight);
+    }
+
+    private void setupBaseGui(GTOreDefinition oreDefinition){
+        NonNullList<ItemStack> containedOresAsItemStacks = NonNullList.create();
+        List<BlockState> blocks = oreDefinition.getVeinGenerator().getAllBlocks();
+        List<Integer> chances = oreDefinition.getVeinGenerator().getAllChances();
+        blocks.forEach(block -> containedOresAsItemStacks.add(block.getBlock().asItem().getDefaultInstance()));
+        int n = containedOresAsItemStacks.size();
+        int x = (120 - 18 * n) / 2;
+        for (int i = 0; i < n; i++) {
+            SlotWidget oreSlot = new SlotWidget(new ItemStackTransfer(containedOresAsItemStacks), i, x, 18, false, false);
+            int finalI = i;
+            oreSlot.setOnAddedTooltips((stack, tooltips) ->
+                    tooltips.add(Component.nullToEmpty(LocalizationUtils.format("gtceu.jei.ore_vein_diagram.chance", chances.get(finalI)))));
+            oreSlot.setIngredientIO(IngredientIO.OUTPUT);
+            addWidget(oreSlot);
+            x += 18;
+        }
+    }
+
+    private void setupBaseGui(BedrockFluidDefinition fluid){
+        Fluid storedFluid = fluid.getStoredFluid().get();
+        TankWidget fluidSlot = new TankWidget(
+                new FluidStorage(FluidStack.create(storedFluid, 1000)), 51, 18, false, false);
+        fluidSlot.setIngredientIO(IngredientIO.OUTPUT);
+        addWidget(fluidSlot);
+    }
+
+    private void setupText(){
+        addWidget(new LabelWidget(5, 0,
+                LocalizationUtils.format("gtceu.jei.ore_vein_" + name)));
+        addWidget(new LabelWidget(5, 40,
+                LocalizationUtils.format("gtceu.jei.ore_vein_diagram.spawn_range")));
+        addWidget(new LabelWidget(5, 50, range));
+
+        addWidget(new LabelWidget(5, 60,
+                LocalizationUtils.format("gtceu.jei.ore_vein_diagram.weight", weight)));
+        addWidget(new LabelWidget(5, 70,
+                LocalizationUtils.format("gtceu.jei.ore_vein_diagram.dimensions")));
+        addWidget(new LabelWidget(5, 80, dimensions));
+    }
+
+    private String dimensions() {
+        AtomicInteger counter = new AtomicInteger(0);
+        if (dimensionFilter == null) return "Overworld";
+        return dimensionFilter.stream()
+                .map(dimension -> FormattingUtil.toEnglishName(dimension.location().getPath()))
```

Comment:
lets not toEnglishName this, as it won't be localized. Just do:
```suggestion
                .map(dimension -> dimension.location().toString())
```